src/util/TicTacToeGame.ts

```diff
@@ -0,0 +1,230 @@
+import {
+  Move,
+  Player,
+  SmallBoardInformation,
+  SmallTileInformation,
+  TileInformation,
+  TileValue,
+  Winner,
+} from '../state/AppState';
+import { Point, arePointsEqual } from './Point';
+import { playerToTileValue } from './PlayerToTile';
+
+export class TicTacToeGame {
+  private readonly board: SmallBoardInformation[];
+  private readonly moves: Move[];
+  private currentPlayer: Player;
+
+  constructor(moves: Move[]) {
+    this.board = this.getInitialSmallBoards();
+    this.currentPlayer = Player.Cross;
+    this.moves = [];
+
+    this.applyMoves(moves);
+  }
+
+  applyMove(move: Move) {
+    const boardToChange = this.board.find((board) =>
+      arePointsEqual(board.position, move.boardPosition),
+    );
+    const tileToChange = boardToChange!.tiles.find((tile) =>
```

Comment:
maybe just make like

```
if(!boardToChange){
// never happens theoreticaly or do something here
}
```

instead of always using the '!'

Reply:
Done.